appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -459,41 +421,22 @@ public static Link getResendMobileNumberConfirmationLink(UUID userId)
 		return linkBuilder.withRel("resendMobileNumberConfirmationCode");
 	}
 
-	private static Link getUserLink(String rel, UUID userId, Optional<UUID> requestingUserId, Optional<UUID> requestingDeviceId)
+	private static Link getUserLink(String rel, UUID userId, UUID requestingUserId, Optional<UUID> requestingDeviceId)
 	{
-		String requestingUserIdStr = requestingUserId.map(UUID::toString).orElse(null);
+		String requestingUserIdStr = requestingUserId.toString();
 		String requestingDeviceIdStr = requestingDeviceId.map(UUID::toString).orElse(null);
 		return linkTo(methodOn(UserController.class).getUser(Optional.empty(), null, requestingUserIdStr, requestingDeviceIdStr,
-				null, userId)).withRel(rel).expand(OMITTED_PARAMS);
-	}
-
-	public static Link getPublicUserLink(String rel, UUID userId)
-	{
-		return getUserLink(rel, userId, Optional.empty(), Optional.empty());
+				userId)).withRel(rel).expand(OMITTED_PARAMS);
 	}
 
 	public static Link getPrivateUserLink(String rel, UUID userId, Optional<UUID> requestingDeviceId)
```

Comment:
We could consider to do a global rename of the "private user" term to "own user"

Reply:
That wouldn't be the right rename, as the "private" data is also included in the buddy entity and included as ``UserPrivateDataBaseDto`` in the ``UserDto``, but this method here, I will rename.